### PR TITLE
[Merged by Bors] - fix: bump minor versions (BUG-1172)

### PR DESCRIPTION
**Pull request by DecathectZero**
Another attempt to address memory issues. 

We're noticing it creep up but looking through the heapdump it is still incredibly hard to pinpoint where the particular issue is. Trying to update a lot of dependencies slowly as well.

Reference my comments for the decisions behind each bump.
More major bumps I will do as separate and distinct PRs.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,15 +39,14 @@
     "axios": "^0.21.1",
     "compression": "^1.7.4",
     "connect-timeout": "^1.9.0",
-    "cookie-parser": "^1.4.4",
+    "cookie-parser": "^1.4.6",
     "core-js": "^3.4.8",
     "cors": "^2.8.5",
     "cuid": "^2.1.8",
     "date-fns": "^2.30.0",
-    "dd-trace": "^3.41.0",
```
Comment (DecathectZero) [inline package.json:47]: `dd-trace` absolutely spikes up memory. It doesn't really work with node 16.20.2 so going to remove it for now

**Review by JoshTheDeveloper — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -39,15 +39,14 @@
     "axios": "^0.21.1",
     "compression": "^1.7.4",
     "connect-timeout": "^1.9.0",
-    "cookie-parser": "^1.4.4",
```
Comment (DecathectZero) [inline package.json:42]: benign

File: package.json [hunk 3]
```diff
@@ -39,15 +39,14 @@
...
+    "cookie-parser": "^1.4.6",
     "core-js": "^3.4.8",
     "cors": "^2.8.5",
     "cuid": "^2.1.8",
     "date-fns": "^2.30.0",
-    "dd-trace": "^3.41.0",
-    "dedent": "^0.7.0",
+    "dedent": "^1.5.1",
```
Comment (DecathectZero) [inline package.json:47]: benign.

File: package.json [hunk 4]
```diff
@@ -59,17 +58,17 @@
     "json-stringify-safe": "^5.0.1",
     "lodash": "^4.17.21",
     "mathjs": "^10.6.1",
-    "mongodb": "^6.3.0",
```
Comment (DecathectZero) [inline package.json:62]: mongo's release mentioned fixing promise resolution and cursor errors - there is a performance increase?
https://github.com/mongodb/node-mongodb-native/pull/3992
https://github.com/mongodb/node-mongodb-native/pull/3985
`consolidate signal use and abort promise wrap`

I saw something like this in my heapdump:
Screenshot 2024-03-10 at 3 45 19 PM

File: package.json [hunk 5]
```diff
@@ -59,17 +58,17 @@
...
+    "nestjs-zod": "2.3.3",
+    "node-fetch": "^2.7.0",
     "regenerator-runtime": "^0.13.3",
     "require-from-url": "^3.1.3",
     "slate": "0.72.3",
     "talisman": "^1.1.3",
     "ts-pattern": "^4.0.5",
     "unleash-client": "5.0.0",
     "validator": "^13.7.0",
-    "vm2": "^3.9.17",
```
Comment (DecathectZero) [inline package.json:72]: minor update for security purposes
https://github.com/patriksimek/vm2/releases

but we'll move off of this in the long run

File: package.json [hunk 6]
```diff
@@ -120,7 +119,7 @@
     "lint-staged": "^10.5.3",
     "mocha": "^6.2.3",
     "nyc": "^15.1.0",
-    "prettier": "2.8.4",
+    "prettier": "2.8.8",
```
Comment (DecathectZero) [inline package.json:122]: benign

File: package.json [hunk 7]
```diff
@@ -59,17 +58,17 @@
     "json-stringify-safe": "^5.0.1",
     "lodash": "^4.17.21",
     "mathjs": "^10.6.1",
-    "mongodb": "^6.3.0",
-    "nestjs-zod": "2.1.0",
-    "node-fetch": "^2.6.7",
+    "mongodb": "^6.4.0",
+    "nestjs-zod": "2.3.3",
+    "node-fetch": "^2.7.0",
```
Comment (DecathectZero) [inline package.json:63]: benign and stable
https://github.com/node-fetch/node-fetch/releases

File: package.json [hunk 8]
```diff
@@ -59,17 +58,17 @@
     "json-stringify-safe": "^5.0.1",
     "lodash": "^4.17.21",
     "mathjs": "^10.6.1",
-    "mongodb": "^6.3.0",
-    "nestjs-zod": "2.1.0",
-    "node-fetch": "^2.6.7",
+    "mongodb": "^6.4.0",
+    "nestjs-zod": "2.3.3",
```
Comment (DecathectZero) [inline package.json:62]: benign
https://github.com/risen228/nestjs-zod/releases

**Pull request closed without merging**